tests/testprovider/echo.go

```diff
@@ -20,12 +20,40 @@ import (
 	"context"
 	"fmt"
 
+	pschema "github.com/pulumi/pulumi/pkg/v3/codegen/schema"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
 	rpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
 
 	"google.golang.org/protobuf/types/known/emptypb"
 )
 
+func init() {
+	providerSchema.Resources["testprovider:index:Echo"] = pschema.ResourceSpec{
+		IsComponent: false,
+		ObjectTypeSpec: pschema.ObjectTypeSpec{
+			IsOverlay:   false,
+			Description: "A test resource that echoes its input.",
+			Properties: map[string]pschema.PropertySpec{
+				"echo": {
+					TypeSpec: pschema.TypeSpec{
+						Ref: "pulumi.json#/Resource",
+					},
+					Description: "A resource to echo.",
```

Comment:
Shouldn't this be `Ref: "pulumi.json#/Any"` and the same for the input, to align with existing usage?

https://github.com/pulumi/pulumi/blob/4a012def4b2191e3b98f5fa5e36c1eb3c4c3334e/tests/integration/construct_component/testcomponent-python/echo.py#L10

Reply:
You're right, I didn't realize that the Echo resource was able to echo non-resources.